src/pudl/transform/ferc714.py

```diff
@@ -315,35 +361,223 @@ def _post_process(df: pd.DataFrame, table_name: str) -> pd.DataFrame:
+        are nulled.
+        """
+        # we expect all of these submissions to be from the last Q
+        assert all(xbrl.report_period == "Q4")
+        # first we are gonna null out all of the "EIA" codes that are really just the respondent id
+        code_is_respondent_id_mask = xbrl.eia_code.str.startswith("C") & (
+            xbrl.respondent_id_ferc714_xbrl == xbrl.eia_code
+        )
+        xbrl.loc[code_is_respondent_id_mask, "eia_code"] = pd.NA
+
+        # lets null out some of the eia_code's from XBRL that we've manually culled
+        respondent_id_xbrl_to_bad_eia_code = {
+            "C002422": ["5776"],
+            "C011374": ["8376"],
+            "C002869": ["F720204"],
+            "C002732": ["F720204", "57049, 57050"],
+            "C011420": ["16606"],
+        }
+        for rid_xbrl, bad_eia_codes in respondent_id_xbrl_to_bad_eia_code.items():
+            xbrl.loc[
+                (xbrl.respondent_id_ferc714_xbrl == rid_xbrl)
+                & (xbrl.eia_code.isin(bad_eia_codes)),
+                "eia_code",
+            ] = pd.NA
+        return xbrl
+
+    @staticmethod
+    def convert_into_static_table_xbrl(xbrl: pd.DataFrame) -> pd.DataFrame:
+        """Convert this annually reported table into a skinner, static table.
+
+        The CSV table is entirely static - it doesn't have any reported
+        changes that vary over time. The XBRL table does have start and end
+        dates in it. In order to merge these two sources, we are checking
+        whether or not the shared variables change over time and then
+        converting this table into a non-time-varying table.
+        """
+        # the CSV data does not vary by year, so we need to check if that is
+        # also going to be the case for the XBRL data. we check the eia_codes
+        # during ensure_eia_code_uniqueness. The name is less crucial but we
+        # should still check.
+        assert all(
+            xbrl.groupby(["respondent_id_ferc714_xbrl"])[  # noqa: PD101
+                ["respondent_name_ferc714"]
+            ].nunique()
+            == 1
+        )
+        cols_to_keep = [
+            "respondent_id_ferc714",
+            "respondent_id_ferc714_xbrl",
+            "respondent_name_ferc714",
+            "eia_code",
+        ]
+        # we are not checking whether the respondent_name_ferc714 is exactly
+        # consistent across the submissions so before we grab the one true eia_code
+        # we are going to first sort by report year (descending) so the more recent
+        # name is the name we get
+        return (
+            xbrl.sort_values(["report_year"], ascending=False)[cols_to_keep]
+            .sort_values(["respondent_id_ferc714", "eia_code"])
+            .drop_duplicates(subset=["respondent_id_ferc714"], keep="first")
+        )
+
+    @staticmethod
+    def condense_into_one_source_table(df):
+        """Condense the CSV and XBRL records together into one record.
```

Comment:
Not blocking at all, but in the other two tables we just put this code directly into the `run` function instead of creating it's own function wrapper. NBD, but it would be good to be consistent.

Reply:
OR we could very easily make this a function vs. a class method because I think it's the same for all tables. We could just have a column parameter or something.

Reply:
this isn't a concat step like the other ones. actually this table does get concat-ed first. the output of this table is one record per respondent, so before this we have one from csv and one from xbrl. we've made sure the eia_codes are unique and this step just drop dupes.